zeppelin-jupyter/src/main/java/org/apache/zeppelin/jupyter/JupyterUtil.java

```diff
@@ -78,7 +78,7 @@ public JupyterUtil() {
         .registerSubtype(ExecuteResult.class, "execute_result")
         .registerSubtype(DisplayData.class, "display_data").registerSubtype(Stream.class, "stream")
         .registerSubtype(Error.class, "error");
-    this.markdownProcessor = new PegdownParser();
+    this.markdownProcessor = new FlexmarkParser();
```

Comment:
Should put in another PR ?

Reply:
It could be but Pegdown uses `asm` pages inside it but it's related to the implementation by Jetty. So without it, tests wouldn't be passed properly.

Reply:
I also think it need to remove `markdown` dependency from `jupyter-support` and `zeppelin-zengine`.

Reply:
I have thought about your suggestion and agreed that it should be handled at first. https://issues.apache.org/jira/browse/ZEPPELIN-5527

Reply:
I think we should remove the pegdown parser from the md interpreter.
According to [pegdown README](https://github.com/sirthias/pegdown/blob/master/README.markdown), pegdown has reached the end of its life a long time ago.

Reply:
I have prepared something. #4229

Reply:
Yeap. I think it should be handled as well. By the way, it looks like having redundant dependencies on `zeppelin-jupyter` by having `markdown`.  `JupyterUtil` parser markdown paragraph but I don't think we don't have to depend on the whole `markdown` module for that case. I think we'd better decouple them and copy a small piece of code. #4228